docs/libraries/libraries-with-cli.md

```diff
@@ -0,0 +1,669 @@
+            }
+        }
+    }
+}
+```
+
+A file which does networking would have its `using` directives look like this:
+
+```csharp
+using System;
+#if NET40
+using System.Net;
+#else
+using System.Net.Http;
+using System.Threading.Tasks
+#endif
+```
+
+The compiler is aware of the following names:
+
+```
+NET20  -->  .NET Framework 2.0
+NET35  -->  .NET Framework 3.5
+NET40  -->  .NET Framework 4.0
+NET45  -->  .NET Framework 4.5
+NET451 -->  .NET Framework 4.5.1
+NET452 -->  .NET Framework 4.5.2
+NET46  -->  .NET Framework 4.6
+```
+
+You can now build the project and see the produced artifacts.
+
+Using DNX:
+
+```
+$ dnu restore
+$ dnu build
+```
+
+Using .NET CLI:
+
+```
+$ dotnet restore
+$ dotnet compile
+```
+
+You will now notice three new entries in your `/bin` folder:
+
+```
+/bin
+   /Debug
+      /net40
+      /net45
+      /net451
+      /net452
+      /net46
+      /net462
+```
+
+## How to target a Portable Class Library (PCL)
+
+**NOTE:** This requires Windows and .NET Framework installed on your machine.  Refer to the [Prerequisites](#prerequisites) to get dependencies installed.
+
+Targeting a PCL profile is a bit trickier.  For starters, [reference this list of PCL profiles](http://embed.plnkr.co/03ck2dCtnJogBKHJ9EjY/preview) to make sure you have the correct target.  Hover over the name of each entry for the framework identifier, which you will need.
+
+You will then need to do two things:
+
+1. List the dependencies for each target *inside* of that target.  The "global" `dependencies` section cannot be used.
+2. List the framework assemblies you are using in your code *inside* the PCL target profile section.  This will require, at a minimum, `mscorlib`, `System`, and `System.Core`.  For other APIs (such as `System.Linq`), you will need to find the assembly which corresponds to the API call you're using.  You'll need to then manually list that assembly.
+
+Here is an example targeting PCL Profile 328:
+
+```
+{
+    "frameworks":{
+        ".NETPortable,Version=v4.0,Profile=Profile328":{
+            "frameworkAssemblies":{
+                "mscorlib":"",
+                "System":"",
+                "System.Core":""
+            }
+        }
+    }
+}
+```
+
+This will allow you to target .NET Standard 1.4, .NET Framework 4.0, Windows 8, Windows Phone 8.1, Windows Phone Silverlight 8.1, and Silverlight 5.0.
+
+You can now build.
+
+Using DNX:
+
+```
+$ dnu restore
+$ dnu build
+```
+
+Using .NET CLI:
+
+```
+$ dotnet restore
+$ dotnet compile
+```
+
+You will now notice three new entries in your `/bin/Debug` folder:
+
+```
+/bin
+   /Debug
+      /dotnet55
+      /portable-net40+sl50+netcore45+wpa81+wp8
+```
+
+Each folder contains distinct `.dll` files corresponding to their named target.
+
+## How to cross-compile for .NET Core and .NET Framework
+
+**NOTE:** This requires Windows and .NET Framework installed on your machine.  Refer to the [Prerequisites](#prerequisites) to get dependencies installed.
```

Comment:
It would be great if you could target .NET Framework on Linux (from a compilation standpoint). We should get that added to the backlog if it isn't already. It's just a q of making reference assemblies available outside of using an MSI.